app/models/invenio_rdm_record_converter.rb

```diff
@@ -25,34 +25,39 @@ class InvenioRdmRecordConverter < Sufia::Export::Converter
   MEMOIZED_PERSON_OR_ORG_DATA_FILE = 'memoized_person_or_org_data.txt'
   FUNDING_DATA_FILE = 'app/models/concerns/galtersufia/generic_file/funding_data.txt'
   LICENSE_DATA_FILE = 'app/models/concerns/galtersufia/generic_file/license_data.txt'
+  COLLECTIONS_TO_COMMUNITES_JSON = 'collections_to_communities.json'
   BLANK_FUNDER_SOURCE = {funder: {name: "", identifier: "", scheme: "ror"}, award: {title: "", number: "", identifier: "", scheme: ""}}
 
   @@header_lookup ||= HeaderLookup.new
   @@funding_data ||= eval(File.read(FUNDING_DATA_FILE))
   @@person_or_org_data ||= eval(File.read(MEMOIZED_PERSON_OR_ORG_DATA_FILE))
   @@license_data ||= eval(File.read(LICENSE_DATA_FILE))
+  @@collections_to_communites ||= JSON.parse(File.read(COLLECTIONS_TO_COMMUNITES_JSON))
 
   # Create an instance of a InvenioRdmRecordConverter converter containing all the metadata for json export
   #
   # @param [GenericFile] generic_file file to be converted for export
   def initialize(generic_file=nil, collection_store={})
+    if generic_file.blank?
+      return
+    end
+
+    @generic_file = generic_file
     # communites are necessary to check if the file should be exported
     @collection_store = collection_store
     @communities = list_collections
 
-    if generic_file&.unexportable?(@communities)
+    if generic_file.unexportable?(@communities)
       return
     end
 
-    @generic_file = generic_file
-
     @record = record_for_export
     @file = file_info
     @extras = extra_data
   end
 
   def to_json(options={})
-    options[:except] ||= ["memoized_mesh", "memoized_lcsh", "generic_file", "collection_store"]
+    options[:except] ||= ["memoized_mesh", "memoized_lcsh", "generic_file", "collection_store", "communities"]
```

Comment:
All member variables are being exported so this would include `@communities` which is currently being assigned by `InvenioRdmRecordConverter#list_collections`: 

```
  def list_collections
    collection_paths = []

    @generic_file.collection_ids.each do |collection_id|
      @collection_store[collection_id][:paths].each do |path|
        collection_paths << path
      end
    end

    collection_paths
  end
  ```
  
  I might be wrong on this, but I thought that should be removed, then have "communities" as a field under "metadata" as I implemented in this ticket with the value of the field being the JSON objects @fenekku provided that corresponds to the files collections.

Reply:
Oh, thanks! I'll go ahead and make the change